src/clj/puppetlabs/services/jruby/jruby_core.clj

```diff
@@ -282,3 +289,50 @@
       (cli-ruby! config
         (concat ["-e" (format "load '%s'" url) "--"] args))
       (log/errorf "command %s could not be found in %s" command bin-dir))))
+
+(defmacro with-jruby-instance
+  "Encapsulates the behavior of borrowing and returning a JRubyInstance.
+  Example usage:
+
+  (let [pool-manager-service (tk-app/get-service app :PoolManagerService)
+        pool-context (pool-manager/create-pool pool-manager-service config)]
+    (with-jruby-instance
+      jruby-instance
+      pool-context
+      event-callbacks
+      reason
+
+      (do-something-with-a-jruby-instance jruby-instance)))
+
+  Will throw an IllegalStateException if borrowing a JRubyInstance times out."
+  [jruby-instance pool-context reason event-callbacks & body]
```

Comment:
I think it makes sense to stash `event-callbacks` on the `pool-context` rather than having users pass it in. `borrow-from-pool-with-timeout` and `return-to-pool` both already require having the deref'ed callback atom passed in; I think it makes sense for consistency to also remove it as an argument to them.

Currently the only argument passed to `create-pool` is the config. Do we think that `event-callbacks` should be part of the config, or that `create-pool` should take a second argument? Having it be part of the config feels a little weird, but maybe not that weird?